### ISSUE TITLE
Little doubts about the 'hard' and 'soft' distillation

### ISSUE THREAD

**Issue by CiaoHe** [lucidrains/vit-pytorch#138]
Hi, Phil:

I noticed your implementation of the `hard` and `soft` distillation loss.
https://github.com/lucidrains/vit-pytorch/blob/e5324242be61bcbf433e129e914aa4b4fa1a79a0/vit_pytorch/distill.py#L142-L151

But, compared to what facebook published:
https://github.com/facebookresearch/deit/blob/e6b10b554d17c25c083eda5d5d7505608c6981f8/losses.py#L50-L61
I found in their `hard-distillation loss`, they use student `distillation logits` as input instead of `student logits`. I wonder what's the difference here.

Best

**Comment by lucidrains** [comment]
@CiaoHe you are right HeCao! fixed in the latest :pray: !
